### PR TITLE
Add links to the "Introduction to Clang for Clad contributors" doc

### PR DESCRIPTION
**Pull request by gojakuch**
This commit updates the contributing guidelines and the developers documentation to have link to the newly created doc for beginner contributors.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -69,4 +69,9 @@ review quickly and successfully share several common traits:
 ### Developer Documentation
 
 We have documented several useful hints that usually help when addressing issues
-as they come during developement time in our [developer documentation](https://clad.readthedocs.io/en/latest/user/DevelopersDocumentation.html).
+as they come during developement time in our [developer documentation](https://clad.readthedocs.io/en/latest/user/DevelopersDocumentation.html). 
+
+In case you are interested in seeing the big picture and are just getting 
+started with Clang, there is a document that might interest you: [Introduction to Clang for Clad contributors](https://docs.google.com/document/d/1Dw4BLlWODmR3ohiu_ezswzuJNFQwAA8FRU3WNnvdBrQ/edit?usp=sharing).
```
Comment (vgvassilev) [inline CONTRIBUTING.md:75]: We should have this document in the repo, not linked as google doc. Please take a look how `DevelopersDocumentation.html` is implemented.

Comment (gojakuch) [reply CONTRIBUTING.md:75]: so you mean that this needs to be re-written as a .rst document?

**Comment by gojakuch** [comment]
I'll take a look at why it fails in an hour or so, probably understood the way these docs are built wrong

File: docs/userDocs/source/user/IntroductionToClangForCladContributors.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
+Introduction to Clang for Clad contributors
+**********************************************
+
+Since there’s a lack of official documentation for the LLVM Clang, while the existing docs are not meant to explain how to modify the Clang AST, we are writing this text to help the newcomers get into the technical aspects of Clang useful for contributing to Clad and manipulating the Clang AST in general.
```
Comment (vgvassilev) [inline docs/userDocs/source/user/IntroductionToClangForCladContributors.rst:4]: We generally break at 80 cols.

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Pull request merged**